src/util/RtreeBuild.cpp

```diff
@@ -0,0 +1,1133 @@
+    } else {
+      posWKTStart = wkt.find("LINESTRING(") + 10;
+      posWKTEnd = wkt.find(')', posWKTStart);
+      if (posWKTStart != std::string::npos && posWKTEnd != std::string::npos) {
+        std::string newWkt = wkt.substr(posWKTStart, posWKTEnd - posWKTStart + 1);
+        boundingBox = GetBoundingBoxFromWKT(newWkt);
+      } else {
+        return { };
+      }
+    }
+  }
+
+  return boundingBox;
+}
+
+void Rtree::SaveEntry(Rtree::BoundingBox boundingBox, uint64_t index, std::ofstream& convertOfs) {
+  /**
+     * Save a single entry (which was e.g. converted by ConvertWordToRtreeEntry) to the disk
+   */
+  double minX = boundingBox.min_corner().get<0>();
+  double minY = boundingBox.min_corner().get<1>();
+  double maxX = boundingBox.max_corner().get<0>();
+  double maxY = boundingBox.max_corner().get<1>();
+
+  convertOfs.write(reinterpret_cast<const char *>(&minX), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&minY), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&maxX), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&maxY), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&index), sizeof(uint64_t));
+}
+
+void Rtree::SaveEntryWithOrderIndex(RTreeValueWithOrderIndex treeValue, std::ofstream& convertOfs) {
+  /**
+     * Save a single entry, containing its postion in the x- and y-sorting
+   */
+  double minX = treeValue.box.min_corner().get<0>();
+  double minY = treeValue.box.min_corner().get<1>();
+  double maxX = treeValue.box.max_corner().get<0>();
+  double maxY = treeValue.box.max_corner().get<1>();
+
+  convertOfs.write(reinterpret_cast<const char *>(&minX), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&minY), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&maxX), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&maxY), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&treeValue.id), sizeof(uint64_t));
+  convertOfs.write(reinterpret_cast<const char *>(&treeValue.orderX), sizeof(uint64_t));
+  convertOfs.write(reinterpret_cast<const char *>(&treeValue.orderY), sizeof(uint64_t));
+}
+
+multiBoxGeo Rtree::LoadEntries(const std::string& file) {
+  multiBoxGeo boxes;
+
+  FileReaderWithoutIndex fileReader = FileReaderWithoutIndex(file);
+
+  std::optional<RTreeValue> element = fileReader.GetNextElement();
+  while (element) {
+    boxes.push_back(element.value());
+    element = fileReader.GetNextElement();
+  }
+
+  fileReader.Close();
+
+  return boxes;
+}
+
+multiBoxWithOrderIndex Rtree::LoadEntriesWithOrderIndex(const std::string& file) {
+  multiBoxWithOrderIndex boxes;
+  FileReader fileReader = FileReader(file);
+
+  std::optional<RTreeValueWithOrderIndex> element = fileReader.GetNextElement();
+  while (element) {
+    boxes.push_back(element.value());
+    element = fileReader.GetNextElement();
+  }
+
+  fileReader.Close();
+
+  return boxes;
+}
```

Comment:
These functions should definitely be in the `FileReader.cpp` class.
And let's see whether we even need them with a simpler `FileReader` interface.